Assets/Scripts/UI/HUDManager.cs

```diff
@@ -10,41 +10,33 @@ public class HUDManager : MonoBehaviour
     [SerializeField] private Image drinkHUDImage;        // Drag your DrinkHUDImage here in the inspector
     [SerializeField] private Sprite defaultSprite;       // Sprite to show when no drink is held
 
-    private PlayerInventory playerInventory;
+    [SerializeField] private PlayerInventory playerInventory;
 
     // Start is called before the first frame update
     void Start()
     {
-        // Assuming PlayerInventory is attached to the player
-        playerInventory = FindObjectOfType<PlayerInventory>();
-
-       // Ensures activation of HUD
-        if (playerInventory != null)
+        // Ensure playerInventory is assigned
+        if (playerInventory == null)
         {
-            playerInventory.OnDrinkChanged += UpdateDrinkHUD; 
+            Debug.LogError("PlayerInventory is not assigned in HUDManager!");
+            return;
         }
 
-        // Initialize HUD
-        UpdateDrinkHUD();
-    }
+        // Subscribe to the event
+        playerInventory.OnDrinkChanged += UpdateDrinkHUD;
 
-    // Update is called once per frame
-    void Update()
-    {
-        
+        // Initialize HUD with the current state
+        UpdateDrinkHUD(playerInventory.HasDrink() ? playerInventory.itemsList.Find(item => item.IsDrink) : null);
```

Comment:
I guess it is simpler to write it this way, but in general for optimization, might want to not even call the function with null